src/encode.rs

```diff
@@ -300,7 +316,7 @@ pub trait PublicKeyEncoding: PublicKey {
         let alg = ASN1Block::Sequence(0, vec![oid]);
 
         let bz = self.to_pkcs1()?;
-        let octet_string = ASN1Block::BitString(0, bz.len(), bz);
+        let octet_string = ASN1Block::BitString(0, bz.len() * 8, bz);
```

Comment:
Just a thought, to make the intent of this multiplication by 8 more clear we could do:

```suggestion
        let octet_string = ASN1Block::BitString(0, bz.len() * std::mem::size_of::<u8>(), bz)
```

Even though everybody knows the size of a byte at least it is lest mysterious than just `8`.

Reply:
`std::mem::size_of::<u8>()` is 1, not 8 (it returns bytes, not bits)

Reply:
:man_facepalming: 

Whoops, nevermind, LOL.